$NetBSD: patch-dic_phono_Imakefile,v 1.1 2012/07/03 17:44:39 joerg Exp $

- Tweak canna-dict dirs
- avoid use of ${CPP} as generic preprocessor

--- dic/phono/Imakefile.orig	2015-10-03 19:39:32.000000000 +0000
+++ dic/phono/Imakefile
@@ -27,9 +27,9 @@
 
        ROMAJI_DIC_DEF = -DSHIFT
 
-               DICDIR = $(cannaLibDir)/dic
-            SAMPLEDIR = $(cannaLibDir)/sample
-         SAMPLESRCDIR = $(cannaLibDir)/sample/src
+               DICDIR = $(DicDir)
+            SAMPLEDIR = $(DicDir)/sample
+         SAMPLESRCDIR = $(DicDir)/sample/src
 
     KPDEF = default.kpdef jdaemon.kpdef just.kpdef kaisoku.kpdef kana.kpdef \
             lan5.kpdef matsu.kpdef newjis.kpdef romaji.kpdef sokuon.kpdef \
@@ -54,16 +54,16 @@ AllTarget($(ROMKANA_TABLES))
 
 #ifdef __EMX__
 .kpdef.kp:
-	$(FORCPP) -7 < $< | $(CPP) $(ROMAJI_DIC_DEF) | $(FORCPP) -8 | $(KPDIC) > $@
+	$(FORCPP) -7 < $< | tee ${.TARGET}.before | ${AWK} -f dic-phono-filter.awk | tee ${.TARGET}.after |$(FORCPP) -8 | $(KPDIC) > $@
 
 .kpdef.cbp:
-	$(FORCPP) -7 < $< | $(CPP) $(ROMAJI_DIC_DEF) | $(FORCPP) -8 | $(KPDIC) > $@
+	$(FORCPP) -7 < $< | tee ${.TARGET}.before | ${AWK} -f dic-phono-filter.awk  | tee ${.TARGET}.after | $(FORCPP) -8 | $(KPDIC) > $@
 #else
 .kpdef.kp:
-	$(FORCPP) -7 < $< | $(CPP) $(ROMAJI_DIC_DEF) | $(FORCPP) -8 | $(KPDIC) > `basename $< '.kpdef'`.kp
+	$(FORCPP) -7 < $< | tee ${.TARGET}.before | ${AWK} -f dic-phono-filter.awk  | tee ${.TARGET}.after | $(FORCPP) -8 | $(KPDIC) > `basename $< '.kpdef'`.kp
 
 .kpdef.cbp:
-	$(FORCPP) -7 < $< | $(CPP) $(ROMAJI_DIC_DEF) | $(FORCPP) -8 | $(KPDIC) > `basename $< '.kpdef'`.cbp
+	$(FORCPP) -7 < $< | tee ${.TARGET}.before | ${AWK} -f dic-phono-filter.awk  | tee ${.TARGET}.after | $(FORCPP) -8 | $(KPDIC) > `basename $< '.kpdef'`.cbp
 #endif
 
 MakeDirectories(install,$(DICDIR))
